src/av.mli

```diff
@@ -157,17 +160,17 @@ val reuse_output : input container -> bool -> unit
 val open_output : string -> output container
 (** [Av.open_output filename] open the output file named [filename]. @raise Error if the opening failed. *)
 
-val open_output_format : (output, _) format -> output container
-(** [Av.open_output_format format] open the output [format]. @raise Error if the opening failed. *)
-
-val open_output_format_name : string -> output container
-(** [Av.open_output_format_name name] open the output format of name [name]. @raise Error if the opening failed. *)
+type write_callbacks = {
+  write: bytes -> int -> int -> int;
+  seek : (int -> Unix.seek_command -> int) option
+}
 
+val open_output_stream : (output, _) format -> write_callbacks -> output container
```

Comment:
The decision to use record for callback types wasn't great. I'll change it for the next API breaking bump..